libs/x-content/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java

```diff
@@ -39,6 +43,25 @@ public JsonXContentParser(NamedXContentRegistry xContentRegistry,
         this.parser = parser;
     }
 
+    public JsonXContentParser(
+        NamedXContentRegistry xContentRegistry,
+        DeprecationHandler deprecationHandler,
+        JsonParser parser,
+        RestApiVersion restApiVersion,
+        FilterPath[] include,
+        FilterPath[] exclude
+    ) {
+        super(xContentRegistry, deprecationHandler, restApiVersion);
+        JsonParser filtered = parser;
+        if (exclude != null) {
+            filtered = new FilteringParserDelegate(filtered, new FilterPathBasedFilter(exclude, false), true, true);
```

Comment:
This should probably fail in `hasDoubleWildcard`. Unless I can figure out a way to make it work!

Reply:
I added this test.